cumulusci/util/xml/metadata_tree.py

```diff
@@ -0,0 +1,222 @@
+
+    Note that passing a filename or URL is usually faster than passing an open file or file-like object. However, the HTTP/FTP client in libxml2 is rather simple, so things like HTTP authentication require a dedicated URL request library, e.g. urllib2 or requests. These libraries usually provide a file-like object for the result that you can parse from while the response is streaming in.
+    """
+
+    doc = etree.parse(source)
+    return MetadataElement(doc.getroot())
+
+
+def fromstring(source):
+    """Parse a Metadata Tree from a string"""
+    return MetadataElement(etree.fromstring(source))
+
+
+class MetadataElement:
+    '''A class for representing Metadata in a Pythonic tree.
+
+    After you parse into a MetadataElement tree, you can refer
+    to child elements like this:
+
+    >>> from cumulusci.util.xml import metadata_tree
+    >>> Package.types.name.text
+    'StandardValueSet'
+    >>> Package = metadata_tree.fromstring(b"""<?xml version="1.0" encoding="UTF-8"?>
+    ... <Package xmlns="http://soap.sforce.com/2006/04/metadata">
+    ...     <types>
+    ...         <members>CaseReason</members>
+    ...         <members>CaseStatus</members>
+    ...         <members>CaseType</members>
+    ...         <name>StandardValueSet</name>
+    ...     </types>
+    ...     <version>46.0</version>
+    ... </Package>""")
+    >>> Package.types.name.text
+    'StandardValueSet'
+
+    Or you can refer to them like this:
+
+    >>> Package["types"]["name"].text
+    'StandardValueSet'
+
+    That might be convenient if you had the key names in a variable or if you had to refer to an element name
+    which would clash with a Python keyword (like `import`) or an instance method (like `text`).
+
+    You can refer to members of lists by index:
+
+    >>> Package["types"]["members"][1]
+    <members>CaseStatus</members> element
+
+    There are also methods for finding, appending, inserting and removing nodes, which have their own documentation.
+    '''
+
+    __slots__ = ["_element", "_parent", "_ns", "tag"]
+
+    def __init__(
+        self, element: etree._Element, parent: etree._Element = None,
+    ):
+        assert isinstance(element, etree._Element)
+        assert len(element.nsmap) == 1, "Only one namespace allowed"
+        self._element = element
+        self._parent = parent
+        self._ns = next(iter(element.nsmap.values()))
+        self.tag = element.tag.split("}")[1]
+
+    @property
+    def text(self):
+        if len(self._element):
+            text_node = self._element.find(self._add_namespace("text"))
+            if len(text_node) >= 0:
+                return self._wrap_element(text_node)
+        return self._element.text
+
+    @text.setter
+    def text(self, text):
+        self._element.text = text
+
+    def _wrap_element(self, child: etree._Element):
+        return MetadataElement(child, self._element)
+
+    def _add_namespace(self, tag):
+        return "{%s}%s" % (self._ns, tag)
```

Comment:
I used old-fashioned string interpolation because curly-braces already have a special meaning in namespaces. I think that your suggestion would leave out the curly braces that the namespace wants. Of course I can escape the curly braces but this seemed like the lesser evil by a smidgen.

Reply:
Oops, yeah you're right.